### DISCUSSION TITLE
Alternative geolocation database

### DISCUSSION THREAD

**Discussion by ColdHeat** [plausible/analytics#219] [Anything Else]
I saw Plausible on HN awhile ago and noticed that it used Maxmind databases for ip to country mapping. I had a similar issue myself and did not want to force users to download the database and attribute IP data to Maxmind for basic country dat

I wrote an open source country geoip database that perhaps could be useful: 
https://github.com/geoacumen/geoacumen-country

I am using it in my own projects and intend to make it automatically publish new versions every so often. 

I imagine this is more of something that self-hosters could use but I figured I would mention it.

**Comment by ukutaht** [comment]
Thanks for the suggestion (and sorry for my slow response).

I'm definitely interested in packaging a freely licensed country database with Plausible so people don't have to bring their own. As you probably know, I can't distribute Maxmind's own database easily.

A couple questions:
* Are there any accuracy pitfalls with iptoasn.com ?
* Is there any freely available datasets that go to city level? More granular location data has been requested many times.

**Comment by ColdHeat** [comment]
From my research and discussion with people, leveraging a mapping from an ASN to their IP ranges is more reliable than using the RiR's directly published data. 

I don't know the accuracy of iptoasn or the approach in general. But I'm willing to add more data sources if they can be provided to me. 

As for a city level database, I've been trying to do that myself as well and it basically seems impossible unless you use another company's data (and deal with the licensing) or if you build the dataset yourself.

**Comment by ukutaht** [comment]
Thanks!

I'm thinking maybe it's possible to package your country database with Plausible and ask people to bring their own only if they want city level data.

So I'm in favour of packaging the geoacumen database and leaving a config option open to use a custom database. I don't have much motivation to work on this at the moment as we're already running MaxMind happily. Happy to merge a PR for this, I can't give a timeline for when I can pick it up myself.

**Comment by Findus23** [comment]
Hi,

Just for your information: Since the Maxmind license change at the beginning of 2020, Matomo uses the free db-ip database by default:
https://db-ip.com/db/lite.php

It works like the Maxmind database, but is still licensed under a Creative Commons Attribution 4.0 International License.

**Comment by ukutaht** [comment]
Thanks @Findus23 

I like this option more as it also includes cities, which is a highly requested feature for Plausible.

**Reply by vectorselector** [reply]
I was pleasantly surprised to see `GeoLite2-city` included in the license for `GeoLite2-country` when signing up for Maxmind.
I'm now confused as to what the included geo database `db-ip` is and does. You say that it includes cities, but Plausible doesn't go deeper than countries yet, correct?  Is this the same DB? What am I missing? Is this a matter of developing a UI and accessor methods etc, and the data is already there without switching to the Maxmind DB?

**Reply by ukutaht** [reply]
Hey

> I'm now confused as to what the included geo database db-ip is and does.

It's a free country-level database. The same provider has city level data as well but it's not being used at the moment.

> Is this a matter of developing a UI and accessor methods etc, and the data is already there without switching to the Maxmind DB?

No, currently we only collect and store the country information. We are almost ready to launch city-level data feature which means we will collect, store and show both regions and cities as well.

City-level tracking is currently tested and works with:
1. GeoLite2-City database (free)
2. DB-IP [IP-To-Location database](https://db-ip.com/db/ip-to-location) (€249.90 / year)

Unfortunately it doesn't work with the free DB-IP city-level database because it doesn't return ISO codes for regions.

**Comment by metmarkosaric** [comment]
Cities and regions are now finally live on Plausible following a great demand!

We've been testing a few different geolocation databases over the last few months and settled on https://db-ip.com. It's closest to what we want to achieve but we're still not happy with the way they display some cities in some countries. In some cases, they go to the district/borough level and we'd prefer all visits to be on the city level only (for instance by default DB-IP would display Soho or King's Cross rather than London on the city level).

We've done some manual overrides ourselves for some cities and plan to do more as we discover them. If you find these situations for the cities you're familiar with, please add them [to this thread](https://github.com/plausible/analytics/issues/1582) and we'll take a look in order to manually override them. Thanks!

**Reply by rdvo** [reply]
is there a way to run it on the self hosted one ? Im still not seeing any cities show up in my dashboard

**Reply by ruslandoga** [reply]
@ovived 👋

Yes, city-level geolocation is supported starting from v1.5. And in v2 you only need to set `MAXMIND_LICENSE_KEY` to start using an automatically updated GeoLite2-City DB. Please see "Enable automatic MaxMind GeoLite2 updates" in https://github.com/plausible/analytics/discussions/3132

**Reply by rdvo** [reply]
AH Okay! so i dont need to do anything on my docker file ?

**Reply by ruslandoga** [reply]
Do you mean a Dockerfile or a docker-conpose.yml file? If it's the latter, you'd need to set the env var for the maxmind licence key there

**Reply by zenminimalist** [reply]
@ovived You put them two in your docker-compose.yml file 

```
plausible:
    environment:
      - BASE_URL=https://pa.domain.com
      - SECRET_KEY_BASE=mumstheword
      - MAXMIND_EDITION=GeoLite2-City      
      - MAXMIND_LICENSE_KEY=yourkeyfrommaxmind
```
